lib/getusermedia.js

```diff
@@ -1,35 +1,24 @@
+/* globals navigator */
 'use strict';
 
 /**
- * This function is very similar to <code>navigator.getUserMedia</code> except
- * that it does not use callbacks and returns a Promise for a MediaStream
+ * This function is very similar to <code>navigator.mediaDevices.getUserMedia</code>
+ * except that if no MediaStreamConstraints are provided, then bot audio and video
+ * are requested.
  * @function getUserMedia
  * @param {MediaStreamConstraints} [constraints={audio:true,video:true}] - the
- *   MediaStreamConstraints object specifying what kind of LocalMediaStream to
+ *   MediaStreamConstraints object specifying what kind of MediaStream to
  *   request from the browser (by default both audio and video)
  * @returns Promise<MediaStream>
  */
 function getUserMedia(constraints) {
-  return new Promise(function getUserMediaPromise(resolve, reject) {
-    _getUserMedia(constraints || { audio: true, video: true }, resolve, reject);
-  });
-}
-
-function _getUserMedia(constraints, onSuccess, onFailure) {
-  if (typeof window !== 'undefined' && typeof navigator !== 'undefined') {
-    if (typeof navigator.mediaDevices === 'object' &&
-        typeof navigator.mediaDevices.getUserMedia === 'function') {
-      navigator.mediaDevices.getUserMedia(constraints).then(onSuccess, onFailure);
-      return;
-    } else if (typeof navigator.webkitGetUserMedia === 'function') {
-      navigator.webkitGetUserMedia(constraints, onSuccess, onFailure);
-      return;
-    } else if (typeof navigator.mozGetUserMedia === 'function') {
-      navigator.mozGetUserMedia(constraints, onSuccess, onFailure);
-      return;
-    }
+  if (typeof navigator === 'object'
```

Comment:
we should encapsulate this into a function `isWebRTCSupported` and use the function here. I would also export this function so that same code can be used from twilio-video.

Reply:
@makarandp0 Since this module depends on a specific API, its OK to check on only that API.